### DISCUSSION TITLE
Red and dangerous button timing function

### DISCUSSION THREAD

**Discussion by saumon-brule** [ppy/osu#28374] [Q&A] [answered]
Hi, I'm really sorry to start a discussion about something that is really not an issue but I would really like to have an answer (if possible of course). So, as you may know, the red and dangerous buttons on OSU lazer (like deleting scores or quit editing without saving) have a behavior of "loading" or "charging". I would like to get the timing function of the red filling the button but I have no idea of where it is in the code. I could maybe dive into the repo but I don't know a lot about C# and it would take me hours just for this.
So if you have an answer or even just the file (or thefolder maybe) where it is written I would be very happy. If you don't, it's okay it isn't a big problem.
TY ❤️

**Comment by brandondong** [comment, marked as answer]
Relevant file should be https://github.com/ppy/osu/blob/f1689d542eeed350dbb90cb512e4468d4d6bdcfb/osu.Game/Graphics/Containers/HoldToConfirmContainer.cs#L91
